src/Lykke.AzureStorage/Tables/AzureTableStorage.cs

```diff
@@ -435,6 +435,46 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
+        /// <inheritdoc/>
+        public async Task<bool> InsertOrReplaceAsync(T entity, Func<T, bool> replaceCondition)
+        {
+            while (true)
+            {
+                try
+                {
+                    var table = await GetTableAsync();
+                    var existingEntity = await GetDataAsync(entity.PartitionKey, entity.RowKey);
+
+                    if (existingEntity != null)
+                    {
+                        if (!replaceCondition(existingEntity))
+                        {
+                            return false;
+                        }
+
+                        entity.ETag = existingEntity.ETag;
+
+                        await table.ExecuteAsync(TableOperation.Replace(entity), GetRequestOptions(), null);
+                    }
+                    else
+                    {
+                        await table.ExecuteAsync(TableOperation.Insert(entity), GetRequestOptions(), null);
+                    }
+
+                    return true;
+                }
+                catch (StorageException e) when (e.RequestInformation.HttpStatusCode == 412)
```

Comment:
Multiple catches can be merged into one with useing || for catch condition:
e.RequestInformation.HttpStatusCode == 412 || e.RequestInformation.HttpStatusCode == 409 || e.RequestInformation.HttpStatusCode == 404

Reply:
It's a mater of taste, but ok, I have changed